### PR TITLE
fixed Entropy Creeper explosion in water

### PR DESCRIPTION
**Pull request by pixar02**
Solves #65 
- Added Check for empty blockList (you never know)
- Override EffectiveResistance of Entity (no resistance pls, `0f` )

### CODE REVIEW

**Comment by JoeyDP** [comment]
Can you explain what caused the crash? And what does effectiveResistance of an entity do?

**Review by BertJorissen — APPROVED** [review]

**Comment by BertJorissen** [comment]
I have no idea what the effectiveResistance does, but it seems that the bug was caused by the empty list. Your additional check on line 215 seems to solve it.

**Comment by pixar02** [comment]
> Can you explain what caused the crash? And what does effectiveResistance of an entity do?

The override allows the "fake explosion" to actually "destroy" water. We basically make a new explosion, so we can still effect the behavior of the explosion (the physics).
The way I understand the method is it acts as a power scaler for resistance of blocks. (water doesn't explode)

Take a look at the Creeper explode() method, there is a float f that is set to `1f` or `2f` it scales the explosion power.

I think the Effectiveness scales the resistance of blocks. If it stayed `1f` water wouldn't be counted as an affected block since it's resistaned to the explosion. Since we filter out what blocks are effected with `isShuffleable()` we don't need the basic check of resistance.

File: src/main/java/be/uantwerpen/scicraft/entity/EntropyCreeperEntity.java [hunk 1]
```diff
@@ -196,8 +212,8 @@ private void shuffle() {
 
         // Shuffle blocks
         Scicraft.LOGGER.debug("size: " + blocksToShuffle.size());
-        if (world.getGameRules().getBoolean(GameRules.DO_MOB_GRIEFING)) {
-            for(int i = 0; i <= blocksToShuffle.size() * SHUFFLE_PERCENTAGE; i++){
```
Comment (JoeyDP) [inline src/main/java/be/uantwerpen/scicraft/entity/EntropyCreeperEntity.java:200]: Seems like the real problem is over here. The <= has to be a < for the loop to be skipped when it's empty (no explicit check needed). My bad..

**Review by JoeyDP — APPROVED** [review]

**Comment by JoeyDP** [comment]
@pixar02 if you approve this solution, you can merge. Setting the effectiveness to 0 indeed makes more sense for the entropy creeper. Now we can make fountains with it!

**Pull request merged**